contacts/views.py

```diff
@@ -0,0 +1,149 @@
+
+
+@login_required
+@require_http_methods(["GET"])
+def insert(request):
+    return render(request, 'contacts/add_contact.html')
+
+
+@login_required
+@require_http_methods(["POST"])
+def add_contact(request):
+        email = request.POST.get('user_email')
+        first_name = request.POST.get('first_name')
+        last_name = request.POST.get('last_name')
+        note = request.POST.get('note')
+        contact_number = request.POST.get('contact_number')
+        dob = request.POST.get('dob')
+
+        user_id = request.user.id
+        user = User.objects.get(id=user_id)
+
+        contact = Contact(first_name=first_name, last_name=last_name, dob=dob, note=note, user_id=user)
+        contact.save()
+
+        email = Email(email=email, contact_id=contact)
+        email.save()
+
+        number = Number(number=contact_number, contact_id=contact)
+        number.save()
+
+        return HttpResponseRedirect('/user_index/', {'message': 'Deleted Successfully!!'})
+
+
+
+@login_required
+def edit(request, id):
+    contact = Contact.objects.get(id=id)
+    if contact is not None:
+        return render(request, 'contacts/edit_contact.html', {'contact': contact, 'contact_id': id})
+
+
+@require_http_methods(["POST"])
+@login_required
+def delete(request, id):
+    contact = Contact.objects.get(id=id)
+    if contact is not None:
+        emails = Email.objects.filter(contact_id=id)
+        numbers = Number.objects.filter(contact_id=id)
+        if emails is not None and numbers is not None:
+            emails.delete()
+            numbers.delete()
+            contact.delete()
+            return HttpResponseRedirect('/user_index/')
+    return HttpResponse("Not delete id issue while try delete")
+
+
+
+def authenticate_user(request):
+    user_str = str(request.user)
+    if request.user is not None:
+        return HttpResponseRedirect('/user_index/')
+    else:
+        return HttpResponse('%s is not logged in' % user_str)
+
+
+@login_required
+def user_index(request):
+        contact = Contact.objects.filter(user_id=request.user.id)
+        print(contact)
+        return render(request, 'contacts/user_index.html', {'contacts': contact})
+
+
+@login_required
+@require_http_methods(["GET", "POST"])
+def add_email(request, id=None):
+        if request.method == "GET":
+                return render(request, 'contacts/add_email.html', {'contact_id': id})
+        else:
+                email = request.POST.get('email')
+                id = request.POST.get('contact_id')
+
+                contact = Contact.objects.get(id=id)
+                new_email = Email(email=email, contact_id=contact)
+                new_email.save()
+                return HttpResponseRedirect('/user_index/')
+
+@login_required
+@require_http_methods(["GET"])
+def add_number(request, id=None):
+        if request.method == "GET":
+                return render(request, 'contacts/add_number.html', {'contact_id': id})
```

Comment:
I believe the indentation here is a little off. Please verify.

Reply:
yes  was issue in setting on tab editor make 8 spaces sorry for that now i fixed!